dotnet/src/IntegrationTests/Connectors/AzureOpenAI/AzureOpenAIChatCompletion_RequiredFunctionChoiceBehaviorTests.cs

```diff
@@ -0,0 +1,403 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.AzureOpenAI;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+using SemanticKernel.IntegrationTests.TestSettings;
+using Xunit;
+
+namespace SemanticKernel.IntegrationTests.Connectors.AzureOpenAI;
+
+public sealed class AzureOpenAIRequiredFunctionChoiceBehaviorTests : BaseIntegrationTest
+{
+    private readonly Kernel _kernel;
+    private readonly FakeFunctionFilter _autoFunctionInvocationFilter;
+    private readonly IChatCompletionService _chatCompletionService;
+
+    public AzureOpenAIRequiredFunctionChoiceBehaviorTests()
+    {
+        this._autoFunctionInvocationFilter = new FakeFunctionFilter();
+
+        this._kernel = this.InitializeKernel();
+        this._kernel.AutoFunctionInvocationFilters.Add(this._autoFunctionInvocationFilter);
+        this._chatCompletionService = this._kernel.GetRequiredService<IChatCompletionService>();
+    }
+
+    [Fact]
+    public async Task SpecifiedInCodeInstructsConnectorToInvokeKernelFunctionAutomaticallyAsync()
+    {
+        // Arrange
+        this._kernel.ImportPluginFromType<DateTimeUtils>();
+
+        var invokedFunctions = new List<string?>();
+
+        IReadOnlyList<KernelFunction>? SelectFunctions(FunctionChoiceBehaviorFunctionsSelectorContext context)
+        {
+            // Get all function names that have been invoked
+            var invokedFunctionNames = context.ChatHistory
+                .SelectMany(m => m.Items.OfType<FunctionResultContent>())
+                .Select(i => i.FunctionName);
+
+            invokedFunctions.AddRange(invokedFunctionNames);
+
+            if (invokedFunctionNames.Contains("GetCurrentDate"))
+            {
+                return null; // Don't advertise any more functions because the expected function has been invoked.
```

Comment:
nit. Not a very fan of `null` to advertise any more functions. This should be null only if there's no SelectFunctions delegate provided right?

```suggestion
                return []; // Don't advertise any more functions because the expected function.
```

Reply:
In this case it does not really matter. I've changed to empty array instead of null. I also plan to revise nullability of functions in the FunctionChoiceBehaviorConfigurationContext.